### PR TITLE
fix(dropdowns): ensure focus is returned to menu trigger before calling `onChange`

### PR DESCRIPTION
**Pull request by ze-flo**
## Description

This PR bumps `@zendeskgarden/container-menu` to `v0.5.0` with two key changes:

1. `Menu` now returns focus to the trigger before calling `onChange`, improving A11y for implementations that open modals after selection (see https://github.com/zendeskgarden/react-components/pull/1931)
2. `defaultFocusedValue` only applies when opening the menu via keyboard, aligning with [APG examples](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-editor/)

These changes improve accessibility and user experience, especially for keyboard navigation and complex UI interactions.

## 💥 BREAKING CHANGE:

`Menu`: new `restoreFocus` prop (default: `true`) returns focus to trigger after menu interaction. When menu expansion is controlled to allow multiple item selection, set `restoreFocus={false}` and manage trigger focus manually on close.

## Detail
- Bump `@zendeskgarden/container-menu` to `v0.5.0`
- Update specs to reflect focus behavior when using `defaultFocusedValue`
- Adds breaking change detail to v9 migration doc. 

## Checklist

- [ ] ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [ ] ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- [ ] ~~:black_circle: renders as expected in dark mode~~
- [ ] ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [x] :wheelchair: tested for WCAG 2.1 AA accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

File: packages/dropdowns/src/elements/menu/Menu.spec.tsx [hunk 1]
```diff
@@ -148,8 +148,22 @@ describe('Menu', () => {
...
+    expect(trigger).toHaveFocus();
 
+    // close menu
+    await user.click(trigger);
+    expect(item).not.toBeVisible();
+
+    // open menu with keyboard
+    trigger.focus();
+    await user.keyboard(' ');
+    // focus is on the focused item associated with `defaultFocusedValue`
```
Comment (ze-flo) [inline packages/dropdowns/src/elements/menu/Menu.spec.tsx:166]: https://github.com/zendeskgarden/react-containers/pull/660 changed the focus behavior when using `defaultFocusedValue`.

**Review by jzempel — CHANGES_REQUESTED** [review]
The `Menu` API (and Storybook controls) are missing the new `restoreFocus` prop. This needs to be specified under `/types`.

File: docs/migration.md [hunk 2]
```diff
@@ -119,6 +119,9 @@ consider additional positioning prop support on a case-by-case basis.
   - The `v8` version of `@zendeskgarden/react-dropdowns` is no longer maintained and is
     renamed to `@zendeskgarden/react-dropdowns.legacy` in `v9`
 - `Menu`: value `auto` is no longer valid for the `fallbackPlacements` prop.
+- `Menu`: new `restoreFocus` prop (default: `true`) returns focus to trigger
+  after menu interaction. To keep the dropdown open on selection,
+  set `restoreFocus={false}` and manage focus manually.
```
Comment (jzempel) [inline docs/migration.md:124]: Not sure I understand this. Wouldn't the `isExpanded` controlled prop be the mechanism for keeping the menu open?

Comment (ze-flo) [reply docs/migration.md:124]: I was referring to cases when a users chooses to keep the menu open after selecting an item within an `ItemGroup` of type `checkbox`. This [demo](https://66e49e3720ee94778c2a11a5--zendeskgarden-react-containers.netlify.app/?path=/story/packages-menu--controlled-managed-focus) showcases this pattern. 

It probably makes more sense to rephrase it as such:

```
`Menu`: new `restoreFocus` prop (default: `true`) returns focus to trigger
  after menu interaction. If keeping the dropdown open on selection is required,
  set `restoreFocus={false}` and manage focus manually.
```

What do you think?

Comment (jzempel) [reply docs/migration.md:124]: This still makes it sound like there is a causal relationship between keeping the menu open and setting `restoreFocus` to false.

Comment (jzempel) [inline docs/migration.md:124]:
```suggestion
- `Menu`
  - value `auto` is no longer valid for the `fallbackPlacements` prop.
  - new `restoreFocus` prop (default: `true`) returns focus to trigger
    after menu interaction. When menu expansion is controlled to allow
    multiple item selection, set `restoreFocus={false}` and manage trigger
    focus manually on close.
```

Comment (ze-flo) [reply docs/migration.md]: Couldn't have wrote it better myself. Thank you! 💯

**Review by jzempel — APPROVED** [review]
🚢

**Pull request merged**